harness/features/initialize.local.test.ts

```diff
@@ -0,0 +1,182 @@
+import {
+    getConnectionStringForProxy,
+    forEachSDK,
+    createClient,
+    callOnClientInitialized,
+    describeIf
+} from '../helpers'
+import { v4 as uuidv4 } from 'uuid'
+import { getServerScope } from '../mockServer'
+import { Capabilities, SDKCapabilities } from '../types'
+
+jest.setTimeout(10000)
+
+describe('Initialize Tests - Local', () => {
+    const harnessBaseURL = `http://host.docker.internal:${global.__MOCK_SERVER_PORT__}`
+    const scope = getServerScope()
+
+    forEachSDK((name: string) => {
+        const capabilities: string[] = SDKCapabilities[name]
+        const sdkKey = 'server_SDK_KEY'
+        let url: string
+
+        beforeAll(async () => {
+            url = getConnectionStringForProxy(name)
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            it('should error when SDK key is missing', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(url, clientId)
+                const responesBody = await response.json()
+
+                expect(responesBody.exception).toEqual(
+                    'Missing environment key! Call initialize with a valid environment key'
+                )
+            })
+
+            it('should error when SDK key is invalid', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(url, clientId, 'invalid key')
+                const responesBody = await response.json()
+
+                expect(responesBody.exception).toEqual(
+                    'Invalid environment key provided. Please call initialize with a valid server environment key'
+                )
+            })
+
+            it('initializes correctly on valid data', async () => {
+                const clientId = uuidv4()
+                const response = await createClient(url, clientId, sdkKey)
```

Comment:
since theres no baseURLOverride in here, the proxy server will try and make real requests to our prod api instead of the mock server

Reply:
good catch